chain/client/src/client.rs

```diff
@@ -958,9 +962,11 @@ impl ClientActor {
                 highest_height,
                 &self.network_info.most_weight_peers
             ));
-            // Only body / state sync if header height is latest.
+            // Only body / state sync if header height is close to the latest.
             let header_head = unwrap_or_run_later!(self.chain.header_head());
-            if header_head.height == highest_height {
+            if highest_height <= self.config.block_header_fetch_horizon
```

Comment:
What is this change for

Reply:
To stop from "always catching up" problem, as while we fetching headers - new headers arrive.
Need to make this heuristic a bit smarter, but it's mostly activates in the case of fast blocks like 10-20ms time.